operational_analysis/methods/quality_check_automation.py

```diff
@@ -0,0 +1,331 @@
+        
+        self._df = df
+        self._ws = ws_field
+        self._w = power_field
+        self._t = time_field
+        self._id = id_field
+        self._freq = freq
+        self._lat_lon = lat_lon
+
+        if self._id==None:
+            self._id = 'ID'
+            self._df['ID'] = 'Data'
+
+    @logged_method_call
+    def run(self):
+        """
+        Run the QC analysis functions in order by calling this function.
+        
+        Args:
+            (None)
+            
+        Returns:
+            (None)
+        """
+
+        logger.info("Identifying Time Duplications")
+        self.dup_time_identification()
+        logger.info("Identifying Time Gaps")
+        self.gap_time_identification()
+        logger.info("Evaluating timezone deviation from UTC")
+        self._wtk_ws_diurnal = self.ws_diurnal_prep()
+        self.corr_df_calc()
+        logger.info("Isolating Extrema Values")
+        self.max_min()
+        logger.info("QC Diagnostic Complete")
+
+        
+    def dup_time_identification(self):
+        """
+        This function identifies any time duplications in the dataset.
+
+        Args:
+        (None)
+
+        Returns:
+        (None)
+        """
+        self._time_duplications = self._df.loc[self._df.duplicated(subset= [self._id, self._t]), self._t]
+
+    def gap_time_identification(self):
+        """
+        This function identifies any time gaps in the dataset.
+
+        Args:
+        (None)
+
+        Returns:
+        (None)
+        """
+        self._time_gaps = timeseries.find_time_gaps(self._df[self._t], freq=self._freq)
+
+    def indicesForCoord(self,f):
+
+        """
+        This function finds the nearest x/y indices for a given lat/lon. 
+        Rather than fetching the entire coordinates database, which is 500+ MB, this
+        uses the Proj4 library to find a nearby point and then converts to x/y indices.
+        This function relies on the Wind Toolkit HSDS API.
+        Args:
+        f (h5 file): file to be read in
+        Returns: x and y coordinates corresponding to a given lat/lon as a tuple
+        """
+
+        dset_coords = f['coordinates']
+        projstring = """+proj=lcc +lat_1=30 +lat_2=60 
+                    +lat_0=38.47240422490422 +lon_0=-96.0 
+                    +x_0=0 +y_0=0 +ellps=sphere 
+                    +units=m +no_defs """
+        projectLcc = Proj(projstring)
+        origin_ll = reversed(dset_coords[0][0])  # Grab origin directly from database
+        origin = projectLcc(*origin_ll)
+
+        lat, lon = self._lat_lon
+        coords = (lon, lat)
+        coords = projectLcc(*coords)
+        delta = np.subtract(coords, origin)
+        ij = [int(round(x/2000)) for x in delta]
+        return tuple(reversed(ij))
+
+    def ws_diurnal_prep(self, start_date = '2007-01-01', end_date = '2013-12-31'):
+
+        """
+        This method returns a Pandas Series corresponding to hourly average windspeeds
```

Comment:
This does a lot more than just return a pandas series. They key is that It links into the WTK data on AWS as the data source, grabs wind speed data, and calculates diurnal hourly averages.  I think it'd be useful to expand this docstring to include a more detailed description.